iree/compiler/Codegen/LLVMGPU/LLVMGPULowerExecutableTarget.cpp

```diff
@@ -74,31 +107,40 @@ void LLVMGPULowerExecutableTargetPass::runOnOperation() {
   // is fine.
   llvm::StringMap<IREE::HAL::ExecutableEntryPointOp> entryPoints =
       getAllEntryPoints(moduleOp);
-  Optional<IREE::Codegen::DispatchLoweringPassPipeline> passPipeline;
+  Optional<IREE::Codegen::TranslationInfoAttr> translationInfo;
   for (auto &it : entryPoints) {
     auto entryPointOp = it.second;
-    if (IREE::Codegen::TranslationInfoAttr translationInfo =
+    if (IREE::Codegen::TranslationInfoAttr currTranslationInfo =
             getTranslationInfo(entryPointOp)) {
-      IREE::Codegen::DispatchLoweringPassPipeline currPipeline =
-          translationInfo.getDispatchLoweringPassPipeline();
-      if (passPipeline) {
-        if (currPipeline != passPipeline.getValue()) {
-          moduleOp.emitError(
-              "unhandled compilation of entry point function with different "
-              "pass pipelines within a module");
-          return signalPassFailure();
+      if (translationInfo) {
+        if (currTranslationInfo != translationInfo.getValue()) {
+          moduleOp.emitOpError(
+              "unhandled compilation of entry point functions with different "
+              "translation info");
+        }
+
+        // Verify the properties of each entry point based on the target
+        // pipeline.
+        Optional<mlir::ArrayAttr> workgroupSizeAttr =
+            entryPointOp.workgroup_size();
+        if (workgroupSizeAttr.hasValue()) {
+          LogicalResult verificationStatus = verifyEntryPoint(
+              moduleOp, currTranslationInfo, workgroupSizeAttr.getValue());
+          if (failed(verificationStatus)) {
+            return signalPassFailure();
+          }
```

Comment:
Why not move all the logic in the verifyEntryPoint function?
Have something like:
```
if(failed(verifyEntryPoint(
              moduleOp, currTranslationInfo, entryPointOp)) {
              return signalPassFailure();
}
```